libafl/src/stages/sync.rs

```diff
@@ -92,54 +93,51 @@ where
         state: &mut Self::State,
         manager: &mut EM,
     ) -> Result<(), Error> {
-        log::debug!("Syncing from disk: {:?}", self.sync_dir);
         let last = state
             .metadata_map()
             .get::<SyncFromDiskMetadata>()
             .map(|m| m.last_time);
 
-        if let (Some(max_time), mut new_files) = self.load_from_directory(None, &last)? {
-            if last.is_none() {
-                state
-                    .metadata_map_mut()
-                    .insert(SyncFromDiskMetadata::new(max_time, new_files));
-            } else {
-                state
-                    .metadata_map_mut()
-                    .get_mut::<SyncFromDiskMetadata>()
-                    .unwrap()
-                    .last_time = max_time;
-                state
-                    .metadata_map_mut()
-                    .get_mut::<SyncFromDiskMetadata>()
-                    .unwrap()
-                    .left_to_sync
-                    .append(&mut new_files);
+        if let Some(last) = last {
+            if (last + self.interval) < current_time() {
```

Comment:
Isn't this the opposite?
You want to perform this stage once every while,
shouldn't this be current_time() < (last + self.interval)

Reply:
@R9295

Reply:
Yeah, that's why it returns ``Ok(())`` if last + self.interval < current_time
https://github.com/R9295/LibAFL/blob/d9ca434207e1828b7512e5046175ed3a8c33f307/libafl/src/stages/sync.rs#L101-L104

Reply:
But instead of the early return, I can wrap it in a different clause

Reply:
what i mean is if you want early return if not enough time has passed 
then it should be
`(last + self.interval) > current_time()`
right?

Reply:
Ah yes, you're right. Whoops

Reply:
yeah it's confusing
maybe do something like
```
let cur = current_time();
if cur.checked_sub(last) < self.last {
 // early return
}
```